### PR TITLE
Add LengthDelimited struct to parse <len, "bytes"> payloads.

### PR DESCRIPTION
**Pull request by ijager**
This PR adds a struct to help parsing length delimited payloads. There are currently a few assumptions:

- The length delimited payload is at the end of a response. This is because I have not found a way to communicate back the number of eaten bytes used for parsing to the Deserializer.
- The payload is "quoted" but the quotes do not count for the length. This is how I have seen it on Quectel modems.

I still need to do more tests but I wouldn't mind some early feedback.

- I have commandeered `deserialize_tuple` as it was not used. I couldn't use `deserialize_bytes` as it clips the slice at the first `,`.
- In `deserialize_tuple` we set ` self.index = self.slice.len();` which causes `LengthDelimited` to only work at the end of the response.

### CODE REVIEW

**Comment by MathiasKoch** [comment]
Haven't looked much yet, but to your concern on `deserialize_bytes`, I would think you can use that after https://github.com/BlackbirdHQ/atat/pull/182? Doesn't that address exactly that issue?

**Comment by ijager** [comment]
> Haven't looked much yet, but to your concern on `deserialize_bytes`, I would think you can use that after #182? Doesn't that address exactly that issue?

No I don't think so, because it doesn't provide any bytes after the first comma:

https://github.com/BlackbirdHQ/atat/blob/38e0322aac81201b769afc7b63817d8afa795b4a/serde_at/src/de/mod.rs#L493-L510

File: serde_at/src/de/length_delimited.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        E: serde::de::Error,
+    {
+        v.iter()
+            .position(|&c| c == b',')
+            .ok_or_else(|| de::Error::custom("expected a comma"))
+            .and_then(|pos| {
+                let len = parse_len(&v[0..pos])
+                    .map_err(|_| de::Error::custom("expected an unsigned int"))?;
+                // +1 to skip the comma after the length.
+                let start = pos + 1 + 1; // extra +1 to remove outer quotes
```
[the remark this answers is not in the document]
Comment (ijager) [reply serde_at/src/de/length_delimited.rs:59]: Yeah makes sense. I will add that.

**Review by MathiasKoch — APPROVED** [review]
Awesome! 👍 
Thank you very much!

**Pull request merged**